zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -183,6 +190,335 @@ pub(crate) fn get_legacy_transparent_address<P: consensus::Parameters>(
     Ok(None)
 }
 
+/// Returns the transparent address index at the start of the first gap of at least `gap_limit`
+/// indices in the given account, considering only addresses derived for the specified key scope.
+///
+/// Returns `Ok(None)` if the gap would start at an index greater than the maximum valid
+/// non-hardened transparent child index.
+pub(crate) fn find_gap_start(
+    conn: &rusqlite::Connection,
+    account_id: AccountRef,
+    key_scope: KeyScope,
+    gap_limit: u32,
+) -> Result<Option<NonHardenedChildIndex>, SqliteClientError> {
+    match conn
+        .query_row(
+            r#"
+            WITH offsets AS (
+                SELECT
+                    a.transparent_child_index,
+                    LEAD(a.transparent_child_index)
+                        OVER (ORDER BY a.transparent_child_index)
+                        AS next_child_index
+                FROM v_address_first_use a
+                WHERE a.account_id = :account_id
+                AND a.key_scope = :key_scope
+                AND a.transparent_child_index IS NOT NULL
+                AND a.first_use_height IS NOT NULL
+            )
+            SELECT
+                transparent_child_index + 1,
+                next_child_index - transparent_child_index - 1 AS gap_len
```

Comment:
Add a comment here clarifying that both `next_child_index` and `transparent_child_index` are used, so the unused gap between them is one less than their separation.